platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/PlatformSupport.java

```diff
@@ -57,7 +58,12 @@ public Category getRegularPlatformCategory() {
 
     public void configureSchema(AttributesSchemaInternal attributesSchema) {
         configureCategoryDisambiguationRule(attributesSchema);
-        attributesSchema.addFailureDescriber(NoCompatibleVariantsFailure.class, TargetJVMVersionOnLibraryTooNewFailureDescriber.class);
+    }
+    public static void configureFailureHandler(ResolutionFailureHandler handler) {
+        // TODO: This should not be here.
+        // This failure handler has nothing to do with platforms.
+        // This should live in JavaEcosystemSupport.
```

Comment:
We still cannot move this due to the same constraints. It would be good to move but at the moment it would be too difficult to do in this PR. At least the incorrect location is documented.